src/timeline/js/controllers.js

```diff
@@ -757,11 +754,10 @@ App.controller("TimelineCtrl", function ($scope) {
       }
     }
     // Resize timeline
-    if (furthest_right_edge > $scope.project.duration - min_timeline_padding || $scope.project.duration - furthest_right_edge > max_timeline_padding) {
+    if (furthest_right_edge > $scope.project.duration) {
       if ($scope.Qt) {
-        let new_timeline_length = Math.max(min_timeline_length, furthest_right_edge + min_timeline_padding);
-        timeline.resizeTimeline(new_timeline_length);
```

Comment:
This line causes an update to the project's "duration" attribute, which triggers/propagates the change back `controllers::applyJsonDiff`, which then calls this function again. Infinity!